pkg/experiment/ingester/segment.go

```diff
@@ -499,11 +506,13 @@ func (s *segment) ingest(tenantID string, p *profilev1.Profile, id uuid.UUID, la
 		tenant:  tenantID,
 		service: model.Labels(labels).Get(model.LabelNameServiceName),
 	}
+	ds := s.datasetForIngest(k)
+	ds.needsSymbolization = !hasSymbols(p)
```

Comment:
I think it should be:
```go
if !ds.needsSymbolization {
  // ...
}
```
Otherwise, if the last profile does not need symbols we reset the flag that could be set previously.

As optimization, we probably should detect it just once in the very end at flush time (instead of calling `hasSymbols` for each profile)